### ISSUE TITLE
PSA: Plagiarism in Pull Requests

### ISSUE THREAD

**Issue by arnavb** [OpenGenus/cosmos#4721]
As many of you have noticed, I have come back to maintain this repository after a few months.

While I am reviewing pull requests, I've been noticing many examples of people (I won't say names) plagiarising code from other sources, such as other algorithms repositories or websites.

Obviously, this behavior is intolerable. Copying other's work is a form of disrespect in my eyes and as we can see in our [contributing guidelines](https://github.com/OpenGenus/cosmos/blob/master/.github/CONTRIBUTING.md) and [code of conduct](https://github.com/OpenGenus/cosmos/blob/master/.github/CODE_OF_CONDUCT.md) disrespect to others in this community is unacceptable. As a community built on trust, something like plagiarism makes me lose all trust in you as a contributor. I'm sure we all want this repository of information to grow. 

So all in all: _Please do not plagiarise_. I can't check every website and repository to see if it matches your code, so this will be a system built on honor. If I do see plagiarism, do not expect me to readily merge your pull request (as of now, anything in past is in the past).

The reason I've made this issue is so that it gets a lot of visibility from our contributors (hopefully). 

I would appreciate any comments below as to what action should be taken in order to combat plagiarism.

@OpenGenus/cosmos-maintainers @AdiChat @iendeavor I'd like to know your thoughts on this as well. Perhaps we can add a modification to the code of conduct that warns people against copying other's code?

**Comment by parjanyaacoder** [comment]
Well I think contributors can take only that code from a particular repository which they have contributed in that repository.

**Comment by arnavb** [comment]
@parjanyaacoder You are right, if the code copied from another repository is written by the same person then it should be allowed in.

**Comment by Feliren88** [comment]
What if the contributor provides reference? I think as long as there is citation, it should be fine right? As long as the contributor doesn't claim other people's work as him/her
